kymatio/scattering2d/frontend/torch_frontend.py

```diff
@@ -0,0 +1,108 @@
+__all__ = ['Scattering2DTorch']
+
+import torch
+
+from .base_frontend import Scattering2DBase
+from kymatio.scattering2d.core.scattering2d import scattering2d
+from ...frontend.torch_frontend import ScatteringTorch
+
+
+class Scattering2DTorch(ScatteringTorch, Scattering2DBase):
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False, backend='torch'):
+        ScatteringTorch.__init__(self)
+        Scattering2DBase.__init__(**locals())
+        Scattering2DBase._instantiate_backend(self, 'kymatio.scattering2d.backend.')
+        Scattering2DBase.build(self)
+        Scattering2DBase.create_filters(self)
+
+        self.register_filters()
+
+    def register_filters_helper(self, k, v, n, current_filter):
+        if isinstance(k, int):
```

Comment:
In filters, there are more fields, it's to avoid confusion with what is a filter and what is not.

Reply:
> In filters, there are more fields

Yeah that's probably not a great thing in the first place.

> it's to avoid confusion with what is a filter and what is not.

Then maybe we shouldn't call `register_single_filter` on something that's not a filter?

Reply:
That's right. @MuawizChaudhary maybe we can try to remove this?

Reply:
Let's continue the discussion at https://github.com/kymatio/kymatio/pull/449#discussion_r351494465. Resolving this comment.